### ISSUE TITLE
election failed after meta2 restart

### ISSUE THREAD

**Issue by jkasarherou** [open-io/oio-sds#1274]
On a clean platform, I create a new container, after I shutdown one of the nodes I have:

admin/status

```
{
"127.0.0.1:6007": {
"status": {
"status": 303,
"message": "127.0.0.1:6009"
},
"body": ""
},
"127.0.0.1:6008": {
"status": {
"status": 2,
"message": "127.0.0.1:6008: [errno=111] Connection refused"
},
"body": ""
},
"127.0.0.1:6009": {
"status": {
"status": 200,
"message": "OK"
},
"body": ""
}
}
```

When I restart it, after waiting a few seconds for the request to complete:
admin/status

```
"127.0.0.1:6007": {
"status": {
"status": 303,
"message": "127.0.0.1:6009"
},
"body": ""
},
"127.0.0.1:6009": {
"status": {
"status": 200,
"message": "OK"
},
"body": ""
},
"127.0.0.1:6008": {
"status": {
"status": 500,
"message": "Election failed for 6510E7BBD31B2C41C42EFDA23A93F4F1BE3E171FB5E335583E4CD6C742882580.1.meta2"
},
"body": ""
}
}
```

Election debug

admin/debug

```
"127.0.0.1:6008": {
"status": {
"status": 200,
"message": "OK"
},
"body": {
"local": {
"id": null, [80/1902]
"url": "127.0.0.1:6008",
"state": "NONE"
},
"master": {
"id": null,
"url": null
},
"base": {
"name": "6510E7BBD31B2C41C42EFDA23A93F4F1BE3E171FB5E335583E4CD6C742882580.1",
"type": "meta2",
"zk": "471BE4BF4368BEDA75827B5C713F557228C9D0C134054453503F9E1D4B73BF95"
},
"#": {
"R": 20,
"P": 0,
"V": 0
},
"peers": [
"127.0.0.1:6007",
"127.0.0.1:6009"
],
"log": [
"LEAVING_FAILING:LEAVE_OK:FAILED",
"LEAVING_FAILING:LEFT_SELF:LEAVING_FAILING",
"CHECKING_MASTER:GETVERS_KO:LEAVING_FAILING",
"CHECKING_MASTER:GETVERS_KO:CHECKING_MASTER",
"CHECKING_MASTER:GETVERS_KO:CHECKING_MASTER",
"ASKING:MASTER_OK:CHECKING_MASTER",
"LISTING:LIST_OK:ASKING",
"WATCHING:EXISTS_OK:LISTING",
"CREATING:CREATE_OK:WATCHING",
"NONE:NONE:CREATING",
"LEAVING_FAILING:LEAVE_OK:FAILED",
"LEAVING_FAILING:LEFT_SELF:LEAVING_FAILING",
"CHECKING_MASTER:GETVERS_KO:LEAVING_FAILING",
"CHECKING_MASTER:GETVERS_KO:CHECKING_MASTER",
"CHECKING_MASTER:GETVERS_KO:CHECKING_MASTER",
"ASKING:MASTER_OK:CHECKING_MASTER"
]
}
},
```

The container is stuck in that state, the only way to get it working again is to perform a manual admin/leave.

**Comment by jkasarherou** [comment]
I used current master branch.

**Comment by jfsmig** [comment]
So many fixes in 4.2.x, propagated to 4.3.x, 4.4.x and master
